Vhar 7720 validoi typa arvot

rckopteri opened the pull request.

rckopteri committed:
Luo validointi typa säätiedoille ja sen varmistava testi

rckopteri committed:
Pieni refaktorointi

rckopteri committed:
Validoi työmaapäiväkirjan kaluston arvot

rckopteri committed:
Pieni refaktorointi

rckopteri committed:
Validoi päivystäjän tiedot

rckopteri committed:
Validoi työnjohtajan tiedot

rckopteri committed:
Refaktoroi työnjohtajan validointia

rckopteri committed:
Validoi tiestön toimenpiteen tehtävät ja muut arvot

rckopteri committed:
Validoi tiestön muut toimenpiteet

rckopteri committed:
Validoi viranomaisen avustaminen

rckopteri committed:
Validoi muut kuvaustekstit

rckopteri committed:
Päivitä aineistot vastaamaan validointeja

jpluomajoki-solita commented on src/clj/harja/palvelin/integraatiot/api/tyomaapaivakirja.clj:
```diff
@@ -27,14 +28,191 @@
+(defn validoi-saa [saatiedot]
+  (doseq [s saatiedot
+          :let [saa (:saatieto s)]]
+    (when (or (< (:ilman-lampotila saa) -80) (> (:ilman-lampotila saa) 80))
```
Onko tämä kahdesti? Yksi tässä ja toinen null-checkin kanssa alempana?

jpluomajoki-solita commented on src/clj/harja/palvelin/integraatiot/api/tyomaapaivakirja.clj:
```diff
@@ -27,14 +28,191 @@
+      {:koodi virheet/+puutteelliset-parametrit+
+       :viesti (format "Muiden kirjausten kuvausteksti pitää olla asiallisen mittainen. Oli nyt: %s." (:kuvaus (:muut-kirjaukset data)))})))
+
 ;; TODO: Validoi sisään tuleva data
```
Todo-kommentti on varmaan nyt toteutettu?

rckopteri replied:
Kyllä ja nyt kommentti poistettu.

jpluomajoki-solita reviewed (approved):
Jos oikein ymmärsin,  niin nyt vain ensimmäinen virhe jää kiinni, jos tässä on useita. Onko se haluttu toimintatapa? Mun puolesta kelpaa näinkin, kunhan ajattelin varmistaa.

rckopteri committed:
Merge branch 'develop' into VHAR-7720-validoi-typa-arvot

rckopteri committed:
Merge branch 'develop' into VHAR-7720-validoi-typa-arvot

rckopteri committed:
Muokkaa työmaapäiväkirjan validointivirheet

sellaiseksi, että monesta virheestä tulee vain yksi response string.
Ja päivitä testit vastaamaan sitä.

rckopteri committed:
Merge branch 'develop' into VHAR-7720-validoi-typa-arvot
